ui/static/ui/js/facets.js

```diff
@@ -0,0 +1,42 @@
+define('facets', ['jquery', 'bootstrap', 'icheck'], function($) {
+  'use strict';
+  var constructFacetQuery = function(facetCheckbox) {
+    // Takes a dom element and returns the string
```

Comment:
I did jsdoc strings, and added it to packages, but I didn't set up the build for making the docs other than adding a conf that enables markdown.  I'll plan on doing that later on as there is maybe more of a need.